xarray/core/dataarray.py

```diff
@@ -1377,13 +1378,15 @@ def func(self, *args, **kwargs):
         return func
 
     @staticmethod
-    def _binary_op(f, reflexive=False, join='inner', **ignored_kwargs):
+    def _binary_op(f, reflexive=False, join=None, **ignored_kwargs):
         @functools.wraps(f)
         def func(self, other):
             if isinstance(other, (Dataset, groupby.GroupBy)):
                 return NotImplemented
             if hasattr(other, 'indexes'):
-                self, other = align(self, other, join=join, copy=False)
+                # if user does not specify join, default to OPTIONS['join']
+                how_to_join = join or OPTIONS['join']
```

Comment:
This would trigger an UnboundLocalError:
if join is None:  join = OPTIONS['join']
because join is not defined in the local scope prior to that line.

So the choice is either
1.) if join is None:
         how_to_join = OPTIONS['join']
     else:
         how_to_join = join
or
2.) how_to_join = join or OPTIONS['join']

I opted for (2) for simplicity.  Is the style of (1) generally preferred here?